tests/fixtures/test_force_insert.py

```diff
@@ -0,0 +1,740 @@
+            verbosity=0,
+            app_label="fixtures",
+        )
+        self.assertEqual(
+            Article.objects.get().headline,
+            "Who needs more than one database?",
+        )
+
+    def test_loading_using(self):
+        # Load fixtures 1 and 2. These will load using the 'default' database
+        # identifier explicitly.
+        management.call_command(
+            "loaddata",
+            "--force_insert",
+            "db_fixture_1",
+            verbosity=0,
+            database="default",
+        )
+        management.call_command(
+            "loaddata",
+            "--force_insert",
+            "db_fixture_2",
+            verbosity=0,
+            database="default",
+        )
+        self.assertSequenceEqual(
+            Article.objects.values_list("headline", flat=True),
+            [
+                "Who needs more than one database?",
+                "Who needs to use compressed data?",
+            ],
+        )
+
+    def test_unmatched_identifier_loading(self):
+        # Db fixture 3 won't load because the database identifier doesn't
+        # match.
+        with self.assertRaisesMessage(
+            CommandError, "No fixture named 'db_fixture_3' found."
+        ):
+            management.call_command(
+                "loaddata", "--force_insert", "db_fixture_3", verbosity=0
+            )
+        with self.assertRaisesMessage(
+            CommandError, "No fixture named 'db_fixture_3' found."
+        ):
+            management.call_command(
+                "loaddata",
+                "--force_insert",
+                "db_fixture_3",
+                verbosity=0,
+                database="default",
+            )
+        self.assertQuerySetEqual(Article.objects.all(), [])
+
+    def test_exclude_option_errors(self):
+        """Excluding a bogus app or model should raise an error."""
+        msg = "No installed app with label 'foo_app'."
+        with self.assertRaisesMessage(management.CommandError, msg):
+            management.call_command(
+                "loaddata",
+                "--force_insert",
+                "fixture1",
+                exclude=["foo_app"],
+                verbosity=0,
+            )
+
+        msg = "Unknown model: fixtures.FooModel"
+        with self.assertRaisesMessage(management.CommandError, msg):
+            management.call_command(
+                "loaddata",
+                "--force_insert",
+                "fixture1",
+                exclude=["fixtures.FooModel"],
+                verbosity=0,
+            )
+
+    def test_stdin_without_format(self):
+        """Reading from stdin raises an error if format isn't specified."""
+        msg = "--format must be specified when reading from stdin."
+        with self.assertRaisesMessage(management.CommandError, msg):
+            management.call_command("loaddata", "--force_insert", "-", verbosity=0)
+
+
+class BulkCreateLoadingTests(DumpDataAssertMixin, TestCase):
```

Comment:
I assume most of these tests were copy-pasted from `fixtures.tests` and possibly manually altered to get them passing after passing `bulk_create=True`.

It'd be much better if `tests.FixtureLoadingTests` was adjusted to allow extra options at its class definition time and then you'd have a subclass that specified `options = {"bulk_create": True}`. In its current form it's almost impossible to review.

```diff
diff --git a/tests/fixtures/tests.py b/tests/fixtures/tests.py
index bce55bc355..5e28af8bbc 100644
--- a/tests/fixtures/tests.py
+++ b/tests/fixtures/tests.py
@@ -132,11 +132,18 @@ def _dumpdata_assert(


 class FixtureLoadingTests(DumpDataAssertMixin, TestCase):
+    loaddata_default_options = {"verbosity": 0}
+
+    @classmethod
+    def loaddata(cls, *args, **kwargs):
+        options = {**cls.loaddata_default_options, **kwargs}
+        management.call_command("loaddata", *args, **options)
+
     def test_loading_and_dumping(self):
         apps.clear_cache()
         Site.objects.all().delete()
         # Load fixture 1. Single JSON file, with two objects.
-        management.call_command("loaddata", "fixture1.json", verbosity=0)
+        self.loaddata("fixture1.json")
         self.assertSequenceEqual(
             Article.objects.values_list("headline", flat=True),
             ["Time to reform copyright", "Poker has no place on ESPN"],
```

Reply:
Thank you. I agree but did not want to disturb the original code too much. I will refactor as shown.